app/viewmodels/repository.js

```diff
@@ -53,6 +54,13 @@
     return folders.concat(files);
   };
 
+    Node.prototype.fullPath = function() {
+        if (!this.parent) {
```

Comment:
We don't need to calculate this before it's part of the data we get from GitHub.
`this.source.path`